### ISSUE TITLE
Operation not permitted

### ISSUE THREAD

**Issue by vfarcic** [codedellemc/rexray#626]
I'm running a "Docker for AWS" stack from `https://docker-for-aws.s3.amazonaws.com/aws/beta/aws-v1.12.3-beta10.json`  and RexRay through `basi/rexray` container as a Swarm global service.

```sh
docker service create \
  --name rexray \
  --mode global \
  --log-driver json-file \
  -e AWS_ACCESS_KEY_ID=$AWS_ACCESS_KEY \
  -e AWS_SECRET_ACCESS_KEY=$AWS_SECRET_KEY \
  --mount type=bind,src=/run,dst=/run \
  --mount type=bind,src=/var/lib,dst=/var/lib \
  --mount type=bind,src=/dev,dst=/dev \
  basi/rexray

docker volume create --driver=rexray --opt=size=5 --name=test1

docker run -ti --rm -v test1:/test busybox touch /test/viktor1
```

The output of the last command is:

```
docker: Error response from daemon: VolumeDriver.Mount: {"Error":"open /dev/xvda: operation not permitted"}.
```

**Comment by kacole2** [comment]
@vfarcic we've tried testing this and need to do some further testing. Putting REX-Ray in a container isn't as easy as it sounds. Docker for AWS also makes it very complicated. The problem we are seeing is that you are running your commands from an alpine container. therefore, the right permissions are not being passed to the host system where all the mounting takes place and there are operating system conflicts when trying to use Alpine alongside Debian (for this container's base image).

**Comment by vfarcic** [comment]
@kacole2 Thanks for looking into this. I think that having REX-Ray in "Docker for AWS" would be a great thing.

**Comment by kacole2** [comment]
We agree. Our focus right now is Docker 1.13 plugins architecture. Hoping that will just be a shim to making it work with Docker for AWS without issue (we know that probably won't happen though). This is being tracked in #684

**Comment by koensayr** [comment]
@vfarcic We've released REX-Ray 0.8 with Docker 1.13 support. Perhaps you can give it another shot? 

You will notice the issue @kacole2 mentioned is now closed.
https://github.com/codedellemc/rexray/issues/684 

Feel free to reopen the issue if the problem persists.

**Comment by bvis** [comment]
Just to inform.

I had stability problems in my swarm cluster when I installed the new plugin system (rexray/efs) and tried to reuse existing volumes created via the old system.

I hadn't time to debug it, then I fixed the problem the hard way: create a new cluster with the old system. Everything was fine again.

As I hadn't time to debug it I weren't be able to detect the source of the problem, therefore I have not created any issue, sorry.
